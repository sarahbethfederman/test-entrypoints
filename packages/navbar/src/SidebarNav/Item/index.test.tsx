import * as React from 'react';
import { shallow } from 'enzyme';
import { Item } from './index';
import { ListItem, Link, Button } from './index.style';

describe('Item', () => {
  it('should render a link when there is a href', () => {
    const item = shallow(<Item href="#foobar">Foo Bar</Item>);
    expect(item.find(Link)).toHaveLength(1);
    expect(item.find(Button)).toHaveLength(0);
    expect(item.find(Link).props()).toEqual(
      expect.objectContaining({
        href: expect.any(String),
      })
    );
    expect(item).toMatchSnapshot();
  });

  it('should render a button when there is not a href', () => {
    const item = shallow(<Item onClick={jest.fn()} />);
    expect(item.find(Link)).toHaveLength(0);
    expect(item.find(Button)).toHaveLength(1);
    expect(item.find(Button).props()).toEqual(
      expect.objectContaining({
        onClick: expect.any(Function),
      })
    );
    expect(item).toMatchSnapshot();
  });
});
