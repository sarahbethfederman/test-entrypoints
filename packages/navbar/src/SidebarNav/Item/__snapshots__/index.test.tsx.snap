// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render a button when there is not a href 1`] = `
<styled.li
  role="none"
>
  <styled.button
    depth={0}
    onClick={[MockFunction]}
  />
</styled.li>
`;

exports[`Item should render a link when there is a href 1`] = `
<styled.li
  role="none"
>
  <styled.a
    depth={0}
    href="#foobar"
    role="menu item"
  >
    Foo Bar
  </styled.a>
</styled.li>
`;
